runtime/vm/BufferWriter.hpp

```diff
@@ -316,6 +316,10 @@ class VM_BufferWriter {
 	void
 	writeLEB128PaddedU32(U_32 val)
 	{
+		if ((val >> 28) != 0) {
```

Comment:
The constant belongs on the left, suggest `return` is not needed:
```c
		if (0 != (val >> 28)) {
			_overflow = true;
		} else if (checkBounds(4)) {
```

An equivalent overflow test should be added to `writeLEB128PaddedU64()`.

The bounds check in `writeLEB128()` should be `9`, not `sizeof(U_64)`. Suggest the arguments should just be integer literals as they express the number of bytes of space required.

Reply:
The bounds check in `writeLEB128()` is already 9. If you mean `writeLEB128PaddedU64()`, I think it should be 8?

Reply:
Yes, I see `writeLEB128()` checks for 9 bytes - I don't remember how I saw something different, and yes, the check in `writeLEB128PaddedU64()` should be for 8 bytes.